x-pack/solutions/security/packages/kbn-securitysolution-list-hooks/src/use_read_list_index/index.ts

```diff
@@ -48,3 +60,41 @@ export const useReadListIndex = ({
     error: query.error,
   };
 };
+
+/**
+ * Determines whether an error response from the `readListIndex`
+ * API call indicates that the index is not yet created.
+ */
+export const isIndexNotCreatedError = (err: unknown): err is SecurityAppError => {
+  return isSecurityAppError(err) && isNotFoundError(err);
+};
+
+/**
+ * Parses the result of a 404 error from the read list index API into an actionable response
+ *
+ * This endpoint returns a 404 if either of the underlying indices do
+ * not exist. By default, this causes the http client to throw an error,
+ * but as this is a valid result from the endpoint, we want to catch that error
+ * and present a result that is actionable to the consumer.
+ * @param error The 404 IndexNotCreated error returned from the read list index API
+ */
+export const parseReadIndexResultFrom404Error = (
+  error: SecurityAppError
+): Awaited<ReturnType<typeof readListIndexWithOptionalSignal>> => {
+  if (error.body.message.includes(' and ')) {
```

Comment:
this seems like very generic error and without any context.

If we expect certain wording, maybe it makes sense to include it?
Or use some kind of error code